ts/session/types/OpenGroup.ts

```diff
@@ -0,0 +1,82 @@
+// This is the Open Group equivalent to the PubKey type.
+
+interface OpenGroupParams {
+  server: string;
+  channel: number;
+  conversationId: string;
+}
+
+export class OpenGroup {
+  private static readonly serverRegex = new RegExp(
+    '^([\\w-]{2,}.){1,2}[\\w-]{2,}$'
+  );
+  private static readonly groupIdRegex = new RegExp(
+    '^publicChat:[0-9]*@([\\w-]{2,}.){1,2}[\\w-]{2,}$'
+  );
+  public readonly server: string;
+  public readonly channel: number;
+  public readonly groupId?: string;
+  public readonly conversationId: string;
+
+  constructor(params: OpenGroupParams) {
+    const strippedServer = params.server.replace('https://', '');
+    this.server = strippedServer;
+
+    // Validate server format
+    const isValid = OpenGroup.serverRegex.test(this.server);
+    if (!isValid) {
+      throw Error('an invalid server or groupId was provided');
+    }
+
+    this.channel = params.channel;
+    this.conversationId = params.conversationId;
+    this.groupId = OpenGroup.getGroupId(this.server, this.channel);
```

Comment:
why not give the id as a parameter?

Reply:
You want the ability to be able to generate an ID from just `server` and `channel`. These are the bare minimum to fill out all the params.
If you want to use the id as a parameter, you can use `OpenGroup.from`